### ISSUE TITLE
Can't decode to a field of type derived from a built-in simple type

### ISSUE THREAD

**Issue by kostix** [Thomasdezeeuw/ini#16]
Given a definition

```
import "os"

type cfQueue struct {
    Dir string `ini:"Directory"`
    Mode os.FileMode `ini:"file mode"`
}
```

The package does not unmarshal the matching key/value pair, `file mode = 123`, to the `Mode` field, silently.

I suspect the problem is two-fold.

First, since the type `os.FileMode` is defined as `type FileMode uint32`, when unmarshaling through reflection, you should not just switch on types but rather on the _kinds_ of these types--just what, say, `encoding/json` does. Please see [this playground link](http://play.golang.org/p/al1mdpGveM).

Second, [here](https://github.com/Thomasdezeeuw/ini/blob/master/decode.go#L82) you seem to allow the `setReflectValue()` function to successfully do nothing while I'd expect it to return an error (in the style of "I've no idea how to handle the type of the target field") in the `default` clause of its main `switch` statement.

**Comment by Thomasdezeeuw** [comment]
In the [doc](https://godoc.org/github.com/Thomasdezeeuw/ini#Config.Decode) it mentions which types are supported, custom type, like `os.FileMode`, aren't supported. However I'm a bit surprised that it can't detect the `uint32` underneath. I will take a look at that.

Because `setReflectValue` can't set every value I decided that it was best to skip that value, instead of failing on setting a single configuration value. This is especially useful with configurations that use custom types.

**Comment by Thomasdezeeuw** [comment]
Decode should now be able to set simple custom types like `os.FileMode`, essentially aliases for a builtin type. But I'm still not sure how to handle not being able to handle a certain value.

**Comment by kostix** [comment]
On Mon, 07 Dec 2015 11:08:29 -0800
Thomas de Zeeuw notifications@github.com wrote:

> In the [doc](https://godoc.org/github.com/Thomasdezeeuw/ini#Config.Decode) it
> mentions which types are supported, custom type, like `os.FileMode`,
> aren't supported. However I'm a bit surprised that it can't detect
> the `uint32` underneath. I will take a look at that.

To be honest, while I have read the doc I blindly assumed this package
behaves like encoding/json does.  I beleive I was thinking along the
lines of "...if I were the author I would certainly just snatch most of
the code from stdlib..." :-)

> Because `setReflectValue` can't set every value I decided that it was
> best to skip that value, instead of failing on setting a single
> configuration value. This is especially useful with configurations
> that use custom types.

I don't think I follow the logic.  What I've observed, is that given a
valid key and a matching field, the package silently did nothing.
I mean, if it would parse a valid key/value pair but failed to _locate_
a matching field -- skipping it would be the only sensible way to
proceed (there's nowhere to assign the value).  But I think the case in
point is different: the package knows where the value belongs but is
unable to handle the assignment.  IMO, this should produce an error.

To recap, there's the difference between a key which doesn't matching
anything and a key which does match a field but cannot have its value
properly unmarshaled.

**Comment by Thomasdezeeuw** [comment]
I agree. I think the change I'll make is to return if the field is found, but can't set. And add a special tag tot inform the decoderer not to bother, because it can't be set. Something like this:

```
type myType struct {
  Value myCustomType `ini:"-"`
}
```

PS. I didn't take any of the code from the standard library, but the design is inspired by it.

**Comment by Thomasdezeeuw** [comment]
What do you think?
